hippy/module/pypy_bridge/py_strategies.py

```diff
@@ -97,6 +97,33 @@ def pop(self, w_list, index):
 
         return w_php_val.to_py(interp)
 
+    def getslice(self, w_list, start, stop, step, length):
+        """Adapted from the PyPy implementation"""
+
+        interp = self.space.get_php_interp()
+        w_php_arry_ref = self.unerase(w_list.lstorage)
+        w_php_arry = w_php_arry_ref.deref_temp()
+
```

Comment:
Using only Python:

```
vext01@bencher2:~/research/pyhyp/hippyvm.hippyvm$ cat /tmp/slice_native_py.py 
<?php

$pysrc = <<<EOD
def outer():
    def do_slice(a):
        l = len(a)
        i = 0
        while i < l:
            slice = a[i:]
            i += 1

    class I(object): pass
    s = I()

    ary = [2,4,4,3,4,5,4,"a", "dd", "j", 1.2, "r", "e", "kjfdlkjf", 2, 3, 3, 3,6, 5, s];

    j = 0
    while j < 500000:
        do_slice(ary)
        j += 1
EOD;
compile_py_func_global($pysrc);

outer();

?>
```

Time:

```
vext01@bencher2:~/research/pyhyp/hippyvm.hippyvm$ /opt/multitime/bin/multitime -n5 ./hippy-c /tmp/slice_native_py.py 
===> multitime results
1: ./hippy-c /tmp/slice_native_py.py
            Mean        Std.Dev.    Min         Median      Max
real        0.352       0.063       0.290       0.324       0.443       
user        0.346       0.061       0.288       0.316       0.432       
sys         0.002       0.003       0.000       0.000       0.008       
```

So I think this shows that there is a conversion overhead.